packages/medusa/src/models/cart.ts

```diff
@@ -88,83 +115,91 @@
  *       - claim
  *     default: default
  *   completed_at:
+ *     description: The date with timezone at which the cart was completed.
+ *     nullable: true
  *     type: string
- *     description: "The date with timezone at which the cart was completed."
  *     format: date-time
  *   payment_authorized_at:
+ *     description: The date with timezone at which the payment was authorized.
+ *     nullable: true
  *     type: string
- *     description: "The date with timezone at which the payment was authorized."
  *     format: date-time
  *   idempotency_key:
- *     type: string
  *     description: Randomly generated key used to continue the completion of a cart in case of failure.
+ *     nullable: true
+ *     type: string
  *     externalDocs:
  *       url: https://docs.medusajs.com/advanced/backend/payment/overview#idempotency-key
  *       description: Learn more how to use the idempotency key.
  *   context:
- *     type: object
  *     description: "The context of the cart which can include info like IP or user agent."
+ *     nullable: true
+ *     type: object
  *     example:
  *       ip: "::1"
  *       user_agent: "PostmanRuntime/7.29.2"
  *   sales_channel_id:
- *     type: string
  *     description: The sales channel ID the cart is associated with.
+ *     nullable: true
+ *     type: string
  *     example: null
  *   sales_channel:
  *     description: A sales channel object. Available if the relation `sales_channel` is expanded.
- *     type: object
+ *     nullable: true
+ *     $ref: "#/components/schemas/SalesChannel"
  *   created_at:
+ *     description: The date with timezone at which the resource was created.
  *     type: string
- *     description: "The date with timezone at which the resource was created."
  *     format: date-time
  *   updated_at:
+ *     description: The date with timezone at which the resource was updated.
  *     type: string
- *     description: "The date with timezone at which the resource was updated."
  *     format: date-time
  *   deleted_at:
+ *     description: The date with timezone at which the resource was deleted.
+ *     nullable: true
  *     type: string
- *     description: "The date with timezone at which the resource was deleted."
  *     format: date-time
  *   metadata:
- *     type: object
  *     description: An optional key-value map with additional details
+ *     nullable: true
+ *     type: object
  *     example: {car: "white"}
  *   shipping_total:
- *     type: integer
  *     description: The total of shipping
+ *     type: integer
  *     example: 1000
  *   discount_total:
- *     type: integer
  *     description: The total of discount
+ *     type: integer
  *     example: 800
  *   tax_total:
```

Comment:
Nice catch 🎣 . Done in 003c51d01c90020473d0492cbf69e4314c796990

Reply:
Shouldn't `tax_total`, `item_tax_total`, and `shipping_tax_total` be nullable?

Reply:
Since these are dynamically computed and are not persisted, there is no good reason for any of them to be null. They either exist and have a value or they don't exist. 

In my [Model TS refactor PR](https://github.com/medusajs/medusa/pull/3202/files), I'm actually removing `| null` from these fields. We'll see how that goes and if we need to keep `| null`, I'll add it back to the OAS.